### DISCUSSION TITLE
type must be string, but is object

### DISCUSSION THREAD

**Discussion by xesdiny** [nlohmann/json#3196] [Q&A] [answered]
#### What is the issue you have?
Well,I use `json` to parse the  `variant<binary, string> data` from RTC recv
input params
```
variant<binary, string> data
```
Using 
```
string recv_data_str = get<string>(data);
json message = json::parse(recv_data_str);
```
The `recv_data_str` string result is 
```
line 101: {"type":"playerCount","count":2}
line 101: {"type":"config","peerConnectionOptions":{}}
```
The WARN  is 
```
2021-12-15 20:06:38.021 WARN  [8992] [rtc::impl::Transport::recv@74] [json.exception.type_error.302] type must be string, but is number.
2021-12-15 20:06:38.021 WARN  [8992] [rtc::impl::Transport::recv@74] [json.exception.type_error.302] type must be string, but is object
```
Why this normal json cannot be parsed？

- Compiler: ___ vs2019 cmake
- Operating system: ___  win10
- [*] latest release version 3.10.4
- [*] the `master` branch

**Comment by nlohmann** [comment, marked as answer]
What code did you execute when you got the exception?

**Reply by xesdiny** [reply]
Sorry, isn't exception,is WARN .I’m just surprised why this regular json format parsing of ‘{}’ and numbers will also throw WARN for couldn't parse

**Reply by xesdiny** [reply]
Oh!I found that this problem is a problem when I traverse the JSON content without using the [item() ](https://nlohmann.github.io/json/api/basic_json/items/)method.
before:
```
for each (string key in message) 
{ cout << static_cast<std::string>(message[key]); }
```
after:
```
for (auto it = message.begin(); it != message.end(); ++it)
		{
			std::cout << "key: " << it.key() << ", value:" << it.value() << '\n';
		}
```
